lmdeploy/vl/model/internvl.py

```diff
@@ -106,15 +106,26 @@ def build_model(self):
                 no_split_module_classes=['InternVisionEncoderLayer'],
                 dtype=torch.half)
 
-        self.model = model
+        # We need eval mode to freeze the weights in model, thus,
+        # avoid randomness in inference.
+        self.model = model.eval()
         self.config = config
 
         if getattr(self.config, 'dynamic_image_size', False):
             logger.info('using InternVL-Chat-V1-5 vision preprocess')
-            MEAN = (123.675, 116.28, 103.53)
-            STD = (58.395, 57.12, 57.375)
+            MEAN = (0.485, 0.456, 0.406)
+            STD = (0.229, 0.224, 0.225)
             import torchvision.transforms as T
-            self.transform = T.Compose([T.Normalize(mean=MEAN, std=STD)])
+            from torchvision.transforms.functional import InterpolationMode
+            input_size = self.config.vision_config.image_size
+            self.transform = T.Compose([
+                T.Lambda(lambda img: img.convert('RGB')
+                         if img.mode != 'RGB' else img),
+                T.Resize((input_size, input_size),
+                         interpolation=InterpolationMode.BICUBIC),
+                T.ToTensor(),
+                T.Normalize(mean=MEAN, std=STD)
+            ])
```

Comment:
where does `resize` come from?

Reply:
copy from https://huggingface.co/OpenGVLab/InternVL-Chat-V1-5, but resize may not be necessary because patches have been resized in dynamic_preprocess. The InterpolationMode in T.Resize is BICUBIC(the default sampling mode in PIL.Image.Image.resize is PIL.Image.NEAREST.), so, i choose to keep the original preprocess without any change in order to keep the same processing as in InternVL 1.5 Chat demo.